File: Reversing the equation[4-June-2023]/solution.cpp
// >> C++ CODE <<

string reverseEqn (string s)
{
    int i = s.length() - 1;
        string ans = "";
        
        while (i >= 0) {
            
            int num = s[i] - '0';
                
            int j = i - 1;
            int pw=1;
            while (j >= 0 && s[j] >= '0' && s[j] <= '9') {
                num += (s[j] - '0') * pow(10,pw);
                pw++;
                j -= 1;
            }
        
            ans += to_string(num);
            while(j>=0 and (s[j]<'0' or s[j]>'9')){
                 ans+=s[j];
                 j -= 1;
            }
            
            i = j;
        }
        
       
        
        return ans;
}

File: Reversing the equation[4-June-2023]/solution.java
//  >> JAVA CODE <<

String reverseEqn(String s)
{
    
   int i = s.length() - 1;
        StringBuilder ans = new StringBuilder();
        
        while (i >= 0) {
            int num = (int)(s.charAt(i)-'0');
            
            int j = i - 1;
            int pow=1;
            while (j >= 0 && s.charAt(j)>='0' && s.charAt(j)<='9') {
                num += (int)(s.charAt(j)-'0') * Math.pow(10,pow);
                pow++;
                j--;
            }
        
            ans.append(Integer.toString(num));
            
            while (j >= 0 && (s.charAt(j) < '0' || s.charAt(j) > '9')) {
                ans.append(s.charAt(j));
                j--;
            }
            
            i = j;
        }
        
        return ans.toString();
}
